it/v1/endpoints/CreateAndAmendCharitableGivingControllerISpec.scala

```diff
@@ -0,0 +1,246 @@
+        response.status shouldBe OK
+        response.json shouldBe responseBody
+        response.header("X-CorrelationId").nonEmpty shouldBe true
+      }
+    }
+
+    "return error according to spec" when {
+
+      "validation error" when {
+        s"an invalid NINO is provided" in new Test {
+          override val nino: String = "INVALID_NINO"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(NinoFormatError)
+        }
+        s"an invalid taxYear is provided" in new Test {
+          override val mtdTaxYear: String = "INVALID_TAXYEAR"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(TaxYearFormatError)
+        }
+        s"an invalid /giftAidPayments/totalAmount is provided" in new Test {
+          override val requestJson: JsValue = Json.parse(
+            s"""
+               |{
+               |  "giftAidPayments": {
+               |    "totalAmount": -1
+               |  }
+               |}
+               |""".stripMargin
+          )
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(ValueFormatError.copy(paths = Some(Seq("/giftAidPayments/totalAmount"))))
+        }
+        s"a taxYear with range of greater than a year is provided" in new Test {
+          override val mtdTaxYear: String = "2019-21"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(RuleTaxYearRangeInvalidError)
+        }
+
+        s"a taxYear below 2017-18 is provided" in new Test {
+          override val mtdTaxYear: String = "2017-18"
+          override val desTaxYear: String = "2018"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+            DesStub.onError(DesStub.POST, desUri(desTaxYear), BAD_REQUEST, errorBody("INVALID_ACCOUNTING_PERIOD"))
```

Comment:
The name of the test says that it is testing if the taxYear provided is before `2017-18` but the data provided doesn't include a taxYear which is before the minimum.

Instead it's using the des error to make the error which should have been covered in the later test for downstream service errors.